deepchem/feat/tests/test_smiles_featurizers.py

```diff
@@ -0,0 +1,41 @@
+from unittest import TestCase
+import numpy as np
+from nose.tools import assert_equals
+from deepchem.feat import SmilesToSeq, SmilesToImage
+from deepchem.feat.smiles_featurizers import create_char_to_idx
+import os
+
+
+class TestSmilesFeaturizers(TestCase):
+  """Tests for SmilesToSeq and SmilesToImage featurizers."""
+
+  def setUp(self):
+    """Setup."""
+    pad_len = 5
+    max_len = 35
+    filename = os.path.join(
+        os.path.dirname(__file__), "data", "chembl_25_small.csv")
+    char_to_idx = create_char_to_idx(filename, max_len=max_len)
+    self.feat = SmilesToSeq(
+        char_to_idx=char_to_idx, max_len=max_len, pad_len=pad_len)
+
+  def test_smiles_to_seq_featurize(self):
+    """Test SmilesToSeq featurization."""
+    from rdkit import Chem
+    smiles = ["Cn1c(=O)c2c(ncn2C)n(C)c1=O", "CC(=O)N1CN(C(C)=O)C(O)C1O"]
+    mols = [Chem.MolFromSmiles(smile) for smile in smiles]
+    expected_seq_len = self.feat.max_len + 2 * self.feat.pad_len
+
+    features = self.feat.featurize(mols)
+    assert_equals(features.shape[0], len(smiles))
+    assert_equals(features.shape[-1], expected_seq_len)
+
+  def test_reconstruct_from_seq(self):
+    """Test SMILES reconstruction from features."""
+    smiles = "Cn1c(=O)c2c(ncn2C)n(C)c1=O"
+    from rdkit import Chem
+    mol = Chem.MolFromSmiles(smiles)
+    features = self.feat._featurize(mol)
```

Comment:
Is there a reason you call the internal `_featurize()` instead of the public `featurize()`?

Reply:
No reason in particular. Since this was a single SMILES string, I used the internal `_featurize()` instead of `featurize()`.

Reply:
It's good to use public methods in tests when possible, since otherwise you aren't testing the exact code path that users will be calling.

Reply:
Done